{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "daed7512-7d9a-48ff-bcfb-9f9e80d37d89",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-11-26 20:27:02.514781: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:02.514868: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:02.514889: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:02.527311: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n"
     ]
    }
   ],
   "source": [
    "import os.path as osp\n",
    "import subprocess\n",
    "import os\n",
    "from configs.constants import XMAGICAL_EMBODIMENT_TO_ENV_NAME\n",
    "\n",
    "from absl import app\n",
    "from absl import logging\n",
    "from configs.constants import ALGORITHMS\n",
    "from configs.constants import EMBODIMENTS\n",
    "from torchkit.experiment import string_from_kwargs\n",
    "from torchkit.experiment import unique_id\n",
    "import yaml\n",
    "import random\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7f3ed4f3-bf9f-4f80-85b4-c5f0aa00839a",
   "metadata": {},
   "outputs": [],
   "source": [
    "seeds = [0, 5]\n",
    "device = \"cuda:0\"\n",
    "\n",
    "experiment_name = '/home/user/xirl/exp/same_long_gnn/'\n",
    "embodiment = 'longstick'\n",
    "\n",
    "with open(osp.join(experiment_name, \"config.yaml\"), \"r\") as fp:\n",
    "    kwargs = yaml.load(fp, Loader=yaml.FullLoader)\n",
    "\n",
    "if kwargs[\"algorithm\"] == \"goal_classifier\":\n",
    "    reward_type = \"goal_classifier\"\n",
    "else:\n",
    "    reward_type = \"distance_to_goal\"\n",
    "\n",
    "# Map the embodiment to the x-MAGICAL env name.\n",
    "env_name = XMAGICAL_EMBODIMENT_TO_ENV_NAME[embodiment]\n",
    "\n",
    "logging.info(\"Experiment name: %s\", experiment_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f970db6-f868-44ea-9564-889901d19a83",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-11-26 20:27:19.768186: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:19.768255: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:19.768277: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:19.823514: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-11-26 20:27:20.513351: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:20.515288: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:20.515429: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:20.545994: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-11-26 20:27:20.654521: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:20.654792: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:20.654925: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:20.681622: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-11-26 20:27:20.918432: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:20.921234: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:20.921261: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:20.939799: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-11-26 20:27:20.988495: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
      "2024-11-26 20:27:20.989105: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
      "2024-11-26 20:27:20.989516: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
      "2024-11-26 20:27:21.010856: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2024-11-26 20:27:36.536325: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:36.665287: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:36.666511: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "I1126 20:27:36.684592 124201781991232 train_policy_gnn.py:102] Using device: cuda:0\n",
      "I1126 20:27:36.685116 124201781991232 train_policy_gnn.py:106] RL experiment seed: 3\n",
      "2024-11-26 20:27:37.165327: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.293756: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.308664: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.314775: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.315539: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "I1126 20:27:37.342732 123465493583680 train_policy_gnn.py:102] Using device: cuda:0\n",
      "I1126 20:27:37.343389 123465493583680 train_policy_gnn.py:106] RL experiment seed: 2\n",
      "2024-11-26 20:27:37.433120: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.433955: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.439316: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.443508: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.443721: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "I1126 20:27:37.454956 135284192995136 train_policy_gnn.py:102] Using device: cuda:0\n",
      "I1126 20:27:37.455752 135284192995136 train_policy_gnn.py:106] RL experiment seed: 1\n",
      "I1126 20:27:37.474821 138720457398080 train_policy_gnn.py:102] Using device: cuda:0\n",
      "I1126 20:27:37.475442 138720457398080 train_policy_gnn.py:106] RL experiment seed: 0\n",
      "2024-11-26 20:27:37.578846: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2024-11-26 20:27:37.580954: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:894] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "I1126 20:27:37.600247 123931579619136 train_policy_gnn.py:102] Using device: cuda:0\n",
      "I1126 20:27:37.604267 123931579619136 train_policy_gnn.py:106] RL experiment seed: 4\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.\n",
      "  warnings.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet18_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet18_Weights.DEFAULT` to get the most up-to-date weights.\n",
      "  warnings.warn(msg)\n",
      "W1126 20:27:41.037518 124201781991232 checkpoint.py:108] optimizer in saved checkpoint not in checkpoint to reload. Skipping it.\n",
      "I1126 20:27:41.084982 124201781991232 utils.py:114] Restored model from checkpoint 4001.\n",
      "I1126 20:27:41.087513 124201781991232 utils.py:131] Successfully loaded gnn_emb.pkl from /home/user/xirl/exp/same_long_gnn/gnn_emb.pkl\n",
      "I1126 20:27:41.090305 124201781991232 utils.py:131] Successfully loaded gnn_distance_scale.pkl from /home/user/xirl/exp/same_long_gnn/gnn_distance_scale.pkl\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.\n",
      "  warnings.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet18_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet18_Weights.DEFAULT` to get the most up-to-date weights.\n",
      "  warnings.warn(msg)\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.\n",
      "  warnings.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet18_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet18_Weights.DEFAULT` to get the most up-to-date weights.\n",
      "  warnings.warn(msg)\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:195: UserWarning: \u001b[33mWARN: The result returned by `env.reset()` was not a tuple of the form `(obs, info)`, where `obs` is a observation and `info` is a dictionary containing additional information. Actual type: `<class 'numpy.ndarray'>`\u001b[0m\n",
      "  logger.warn(\n",
      "  0%|          | 0/75000 [00:00<?, ?it/s]/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.\n",
      "  warnings.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet18_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet18_Weights.DEFAULT` to get the most up-to-date weights.\n",
      "  warnings.warn(msg)\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.\n",
      "  warnings.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=ResNet18_Weights.IMAGENET1K_V1`. You can also use `weights=ResNet18_Weights.DEFAULT` to get the most up-to-date weights.\n",
      "  warnings.warn(msg)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading chipmunk for Linux (64bit) [/home/user/mambaforge/envs/env/lib/python3.11/site-packages/pymunk/libchipmunk.so]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:219: DeprecationWarning: \u001b[33mWARN: Core environment is written in old step API which returns one bool instead of two. It is recommended to rewrite the environment with new step API. \u001b[0m\n",
      "  logger.deprecation(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:225: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)\n",
      "  if not isinstance(done, (bool, np.bool8)):\n",
      "  0%|          | 48/75000 [00:01<31:23, 39.79it/s]W1126 20:27:42.497077 123931579619136 checkpoint.py:108] optimizer in saved checkpoint not in checkpoint to reload. Skipping it.\n",
      "I1126 20:27:42.533372 123931579619136 utils.py:114] Restored model from checkpoint 4001.\n",
      "I1126 20:27:42.533928 123931579619136 utils.py:131] Successfully loaded gnn_emb.pkl from /home/user/xirl/exp/same_long_gnn/gnn_emb.pkl\n",
      "I1126 20:27:42.534069 123931579619136 utils.py:131] Successfully loaded gnn_distance_scale.pkl from /home/user/xirl/exp/same_long_gnn/gnn_distance_scale.pkl\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.\u001b[0m\n",
      "  logger.warn(\n",
      "W1126 20:27:42.566832 123465493583680 checkpoint.py:108] optimizer in saved checkpoint not in checkpoint to reload. Skipping it.\n",
      "  0%|          | 53/75000 [00:01<31:37, 39.49it/s]W1126 20:27:42.598835 135284192995136 checkpoint.py:108] optimizer in saved checkpoint not in checkpoint to reload. Skipping it.\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:195: UserWarning: \u001b[33mWARN: The result returned by `env.reset()` was not a tuple of the form `(obs, info)`, where `obs` is a observation and `info` is a dictionary containing additional information. Actual type: `<class 'numpy.ndarray'>`\u001b[0m\n",
      "  logger.warn(\n",
      "  0%|          | 0/75000 [00:00<?, ?it/s]I1126 20:27:42.618360 123465493583680 utils.py:114] Restored model from checkpoint 4001.\n",
      "I1126 20:27:42.618797 123465493583680 utils.py:131] Successfully loaded gnn_emb.pkl from /home/user/xirl/exp/same_long_gnn/gnn_emb.pkl\n",
      "I1126 20:27:42.618937 123465493583680 utils.py:131] Successfully loaded gnn_distance_scale.pkl from /home/user/xirl/exp/same_long_gnn/gnn_distance_scale.pkl\n",
      "W1126 20:27:42.620961 138720457398080 checkpoint.py:108] optimizer in saved checkpoint not in checkpoint to reload. Skipping it.\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.\u001b[0m\n",
      "  logger.warn(\n",
      "I1126 20:27:42.633529 135284192995136 utils.py:114] Restored model from checkpoint 4001.\n",
      "I1126 20:27:42.633944 135284192995136 utils.py:131] Successfully loaded gnn_emb.pkl from /home/user/xirl/exp/same_long_gnn/gnn_emb.pkl\n",
      "I1126 20:27:42.634113 135284192995136 utils.py:131] Successfully loaded gnn_distance_scale.pkl from /home/user/xirl/exp/same_long_gnn/gnn_distance_scale.pkl\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:219: DeprecationWarning: \u001b[33mWARN: Core environment is written in old step API which returns one bool instead of two. It is recommended to rewrite the environment with new step API. \u001b[0m\n",
      "  logger.deprecation(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:225: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)\n",
      "  if not isinstance(done, (bool, np.bool8)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading chipmunk for Linux (64bit) [/home/user/mambaforge/envs/env/lib/python3.11/site-packages/pymunk/libchipmunk.so]\n",
      "Loading chipmunk for Linux (64bit) [/home/user/mambaforge/envs/env/lib/python3.11/site-packages/pymunk/libchipmunk.so]\n",
      "Loading chipmunk for Linux (64bit) [/home/user/mambaforge/envs/env/lib/python3.11/site-packages/pymunk/libchipmunk.so]\n",
      "Loading chipmunk for Linux (64bit) [/home/user/mambaforge/envs/env/lib/python3.11/site-packages/pymunk/libchipmunk.so]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "I1126 20:27:42.673341 138720457398080 utils.py:114] Restored model from checkpoint 4001.\n",
      "I1126 20:27:42.674408 138720457398080 utils.py:131] Successfully loaded gnn_emb.pkl from /home/user/xirl/exp/same_long_gnn/gnn_emb.pkl\n",
      "I1126 20:27:42.676502 138720457398080 utils.py:131] Successfully loaded gnn_distance_scale.pkl from /home/user/xirl/exp/same_long_gnn/gnn_distance_scale.pkl\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:195: UserWarning: \u001b[33mWARN: The result returned by `env.reset()` was not a tuple of the form `(obs, info)`, where `obs` is a observation and `info` is a dictionary containing additional information. Actual type: `<class 'numpy.ndarray'>`\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:174: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed a `seed` instead of using `Env.seed` for resetting the environment random number generator.\u001b[0m\n",
      "  logger.warn(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:187: UserWarning: \u001b[33mWARN: Future gym versions will require that `Env.reset` can be passed `options` to allow the environment initialisation to be passed additional information.\u001b[0m\n",
      "  logger.warn(\n",
      "  0%|          | 0/75000 [00:00<?, ?it/s]/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:195: UserWarning: \u001b[33mWARN: The result returned by `env.reset()` was not a tuple of the form `(obs, info)`, where `obs` is a observation and `info` is a dictionary containing additional information. Actual type: `<class 'numpy.ndarray'>`\u001b[0m\n",
      "  logger.warn(\n",
      "  0%|          | 0/75000 [00:00<?, ?it/s]/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:219: DeprecationWarning: \u001b[33mWARN: Core environment is written in old step API which returns one bool instead of two. It is recommended to rewrite the environment with new step API. \u001b[0m\n",
      "  logger.deprecation(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:225: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)\n",
      "  if not isinstance(done, (bool, np.bool8)):\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:195: UserWarning: \u001b[33mWARN: The result returned by `env.reset()` was not a tuple of the form `(obs, info)`, where `obs` is a observation and `info` is a dictionary containing additional information. Actual type: `<class 'numpy.ndarray'>`\u001b[0m\n",
      "  logger.warn(\n",
      "  0%|          | 0/75000 [00:00<?, ?it/s]/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:219: DeprecationWarning: \u001b[33mWARN: Core environment is written in old step API which returns one bool instead of two. It is recommended to rewrite the environment with new step API. \u001b[0m\n",
      "  logger.deprecation(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:225: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)\n",
      "  if not isinstance(done, (bool, np.bool8)):\n",
      "  0%|          | 2/75000 [00:00<1:03:42, 19.62it/s]/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:219: DeprecationWarning: \u001b[33mWARN: Core environment is written in old step API which returns one bool instead of two. It is recommended to rewrite the environment with new step API. \u001b[0m\n",
      "  logger.deprecation(\n",
      "/home/user/mambaforge/envs/env/lib/python3.11/site-packages/gym/utils/passive_env_checker.py:225: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)\n",
      "  if not isinstance(done, (bool, np.bool8)):\n",
      " 73%|███████▎  | 54999/75000 [1:26:36<31:29, 10.58it/s]    \n",
      " 73%|███████▎  | 54999/75000 [1:26:36<31:29, 10.58it/s]\n",
      " 73%|███████▎  | 54840/75000 [1:26:38<29:33, 11.37it/s]Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 203, in main\n",
      "    eval_stats = evaluate(policy, eval_env, config.num_eval_episodes)\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 71, in evaluate\n",
      "    observation, _, done, info = env.step(action)\n",
      "                                 ^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 75, in step\n",
      "    obs, reward, done, info = self.env.step(action)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 211, in step\n",
      "    imageio.mimsave(filename, self.frames, fps=self.fps)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/v2.py\", line 494, in mimwrite\n",
      "    with imopen(uri, \"wI\", **imopen_args) as file:\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/imopen.py\", line 113, in imopen\n",
      "    request = Request(uri, io_mode, format_hint=format_hint, extension=extension)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 247, in __init__\n",
      "    self._parse_uri(uri)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 412, in _parse_uri\n",
      "    raise FileNotFoundError(\"The directory %r does not exist\" % dn)\n",
      "FileNotFoundError: The directory '/home/user/xirl/exp/same_long_gnn/2/video/eval' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 227, in <module>\n",
      "    app.run(main)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 308, in run\n",
      "    _run_main(main, args)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 254, in _run_main\n",
      "    sys.exit(main(argv))\n",
      "             ^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/utils/pdb_fallback.py\", line 58, in inner_wrapper\n",
      "    return f(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 220, in main\n",
      "    checkpoint_manager.save(i)  # pylint: disable=undefined-loop-variable\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 201, in save\n",
      "    self.checkpoint.save(save_path)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 84, in save\n",
      "    os.rename(tmp_path, save_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp36lz4eas/tmp.ckpt' -> '/home/user/xirl/exp/same_long_gnn/2/checkpoints/54999.ckpt'\n",
      " 73%|███████▎  | 54842/75000 [1:26:38<30:41, 10.95it/s]Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 203, in main\n",
      "    eval_stats = evaluate(policy, eval_env, config.num_eval_episodes)\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 71, in evaluate\n",
      "    observation, _, done, info = env.step(action)\n",
      "                                 ^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 75, in step\n",
      "    obs, reward, done, info = self.env.step(action)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 211, in step\n",
      "    imageio.mimsave(filename, self.frames, fps=self.fps)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/v2.py\", line 494, in mimwrite\n",
      "    with imopen(uri, \"wI\", **imopen_args) as file:\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/imopen.py\", line 113, in imopen\n",
      "    request = Request(uri, io_mode, f"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> /home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py(84)save()\n",
      "-> os.rename(tmp_path, save_path)\n",
      "(Pdb) \n",
      "> /home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py(84)save()\n",
      "-> os.rename(tmp_path, save_path)\n",
      "(Pdb) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ormat_hint=format_hint, extension=extension)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 247, in __init__\n",
      "    self._parse_uri(uri)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 412, in _parse_uri\n",
      "    raise FileNotFoundError(\"The directory %r does not exist\" % dn)\n",
      "FileNotFoundError: The directory '/home/user/xirl/exp/same_long_gnn/1/video/eval' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 227, in <module>\n",
      "    app.run(main)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 308, in run\n",
      "    _run_main(main, args)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 254, in _run_main\n",
      "    sys.exit(main(argv))\n",
      "             ^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/utils/pdb_fallback.py\", line 58, in inner_wrapper\n",
      "    return f(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 220, in main\n",
      "    checkpoint_manager.save(i)  # pylint: disable=undefined-loop-variable\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 201, in save\n",
      "    self.checkpoint.save(save_path)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 84, in save\n",
      "    os.rename(tmp_path, save_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpjszzbcwx/tmp.ckpt' -> '/home/user/xirl/exp/same_long_gnn/1/checkpoints/54999.ckpt'\n",
      " 73%|███████▎  | 54999/75000 [1:26:37<31:29, 10.58it/s]\n",
      " 73%|███████▎  | 54999/75000 [1:26:37<31:29, 10.58it/s]\n",
      " 73%|███████▎  | 54848/75000 [1:26:38<32:45, 10.25it/s]Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 203, in main\n",
      "    eval_stats = evaluate(policy, eval_env, config.num_eval_episodes)\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 71, in evaluate\n",
      "    observation, _, done, info = env.step(action)\n",
      "                                 ^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 75, in step\n",
      "    obs, reward, done, info = self.env.step(action)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 211, in step\n",
      "    imageio.mimsave(filename, self.frames, fps=self.fps)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/v2.py\", line 494, in mimwrite\n",
      "    with imopen(uri, \"wI\", **imopen_args) as file:\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/imopen.py\", line 113, in imopen\n",
      "    request = Request(uri, io_mode, format_hint=format_hint, extension=extension)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 247, in __init__\n",
      "    self._parse_uri(uri)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 412, in _parse_uri\n",
      "    raise FileNotFoundError(\"The directory %r does not exist\" % dn)\n",
      "FileNotFoundError: The directory '/home/user/xirl/exp/same_long_gnn/4/video/eval' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 227, in <module>\n",
      "    app.run(main)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 308, in run\n",
      "    _run_main(main, args)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 254, in _run_main\n",
      "    sys.exit(main(argv))\n",
      "             ^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/utils/pdb_fallback.py\", line 58, in inner_wrapper\n",
      "    return f(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 220, in main\n",
      "    checkpoint_manager.save(i)  # pylint: disable=undefined-loop-variable\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 201, in save\n",
      "    self.checkpoint.save(save_path)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 84, in save\n",
      "    os.rename(tmp_path, save_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpskkyj7k6/tmp.ckpt' -> '/home/user/xirl/exp/same_long_gnn/4/checkpoints/54999.ckpt'\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 203, in main\n",
      "    eval_stats = evaluate(policy, eval_env, config.num_eval_episodes)\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 71, in evaluate\n",
      "    observation, _, done, info = env.step(action)\n",
      "                                 ^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 75, in step\n",
      "    obs, reward, done, info = self.env.step(action)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 211, in step\n",
      "    imageio.mimsave(filename, self.frames, fps=self.fps)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/v2.py\", line 494, in mimwrite\n",
      "    with imopen(uri, \"wI\", **imopen_args) as file:\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/imopen.py\", line 113, in imopen\n",
      "    request = Request(uri, io_mode, format_hint=format_hint, extension=extension)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 247, in __init__\n",
      "    self._parse_uri(uri)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 412,"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> /home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py(84)save()\n",
      "-> os.rename(tmp_path, save_path)\n",
      "(Pdb) \n",
      "> /home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py(84)save()\n",
      "-> os.rename(tmp_path, save_path)\n",
      "(Pdb) \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " in _parse_uri\n",
      "    raise FileNotFoundError(\"The directory %r does not exist\" % dn)\n",
      "FileNotFoundError: The directory '/home/user/xirl/exp/same_long_gnn/0/video/eval' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 227, in <module>\n",
      "    app.run(main)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 308, in run\n",
      "    _run_main(main, args)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 254, in _run_main\n",
      "    sys.exit(main(argv))\n",
      "             ^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/utils/pdb_fallback.py\", line 58, in inner_wrapper\n",
      "    return f(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 220, in main\n",
      "    checkpoint_manager.save(i)  # pylint: disable=undefined-loop-variable\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 201, in save\n",
      "    self.checkpoint.save(save_path)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 84, in save\n",
      "    os.rename(tmp_path, save_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpvfqd_ufw/tmp.ckpt' -> '/home/user/xirl/exp/same_long_gnn/0/checkpoints/54999.ckpt'\n",
      " 73%|███████▎  | 54999/75000 [1:26:49<31:34, 10.56it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 203, in main\n",
      "    eval_stats = evaluate(policy, eval_env, config.num_eval_episodes)\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 71, in evaluate\n",
      "    observation, _, done, info = env.step(action)\n",
      "                                 ^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 75, in step\n",
      "    obs, reward, done, info = self.env.step(action)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/sac/wrappers.py\", line 211, in step\n",
      "    imageio.mimsave(filename, self.frames, fps=self.fps)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/v2.py\", line 494, in mimwrite\n",
      "    with imopen(uri, \"wI\", **imopen_args) as file:\n",
      "         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/imopen.py\", line 113, in imopen\n",
      "    request = Request(uri, io_mode, format_hint=format_hint, extension=extension)\n",
      "              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 247, in __init__\n",
      "    self._parse_uri(uri)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/imageio/core/request.py\", line 412, in _parse_uri\n",
      "    raise FileNotFoundError(\"The directory %r does not exist\" % dn)\n",
      "FileNotFoundError: The directory '/home/user/xirl/exp/same_long_gnn/3/video/eval' does not exist\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 227, in <module>\n",
      "    app.run(main)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 308, in run\n",
      "    _run_main(main, args)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/absl/app.py\", line 254, in _run_main\n",
      "    sys.exit(main(argv))\n",
      "             ^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/utils/pdb_fallback.py\", line 58, in inner_wrapper\n",
      "    return f(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/xirl/train_policy_gnn.py\", line 220, in main\n",
      "    checkpoint_manager.save(i)  # pylint: disable=undefined-loop-variable\n",
      "    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 201, in save\n",
      "    self.checkpoint.save(save_path)\n",
      "  File \"/home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py\", line 84, in save\n",
      "    os.rename(tmp_path, save_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpir9onm79/tmp.ckpt' -> '/home/user/xirl/exp/same_long_gnn/3/checkpoints/54999.ckpt'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> /home/user/mambaforge/envs/env/lib/python3.11/site-packages/torchkit/checkpoint.py(84)save()\n",
      "-> os.rename(tmp_path, save_path)\n",
      "(Pdb) \n"
     ]
    }
   ],
   "source": [
    "# Execute each seed in parallel.\n",
    "procs = []\n",
    "for seed in range(*list(map(int, seeds))):\n",
    "    procs.append(\n",
    "        subprocess.Popen([  # pylint: disable=consider-using-with\n",
    "            \"python\",\n",
    "            \"train_policy_gnn.py\",\n",
    "            \"--experiment_name\",\n",
    "            experiment_name,\n",
    "            \"--env_name\",\n",
    "            f\"{env_name}\",\n",
    "            \"--config\",\n",
    "            f\"configs/xmagical/rl/env_reward.py:{embodiment}\",\n",
    "            \"--config.reward_wrapper.pretrained_path\",\n",
    "            f\"{experiment_name}\",\n",
    "            \"--config.reward_wrapper.type\",\n",
    "            f\"{reward_type}\",\n",
    "            \"--seed\",\n",
    "            f\"{seed}\",\n",
    "            \"--device\",\n",
    "            f\"{device}\",\n",
    "            '--resume',\n",
    "        ]))\n",
    "\n",
    "# Wait for each seed to terminate.\n",
    "for p in procs:\n",
    "    p.wait()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60046f99-a2af-44f4-acf8-f4e4d44b94f1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
